src/Calendar.js

```diff
@@ -26,25 +26,57 @@ export default class Calendar extends Component {
     renderDay: day => day.format('YYYY-MM-D')
   };
 
+  constructor() {
+    super();
+    this.state = {};
+  };
+
+  componentWillMount() {
+    this.componentWillReceiveProps(this.props);
+  };
+
+  componentWillReceiveProps (nextProps) {
+    this.setDate(nextProps.date || moment());
+  };
+
+  setDate (date) {
+    this.setState(state => Object.assign(state, { date }));
```

Comment:
This is questionable, considering date is the only state the calendar has. And especially considering that's the only state I can imagine it having. Feel free to alter this.